datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/GmsGraphQLEngine.java

```diff
@@ -79,6 +126,107 @@ public static GraphQLEngine get() {
+                    .dataFetcher("platform", new AuthenticatedResolver<>(
+                            new LoadableTypeResolver<>(
+                                    DATA_PLATFORM_TYPE,
+                                    (env) -> ((Dataset) env.getSource()).getPlatform().getUrn()))
+                    )
+                    .dataFetcher("downstreamLineage", new AuthenticatedResolver<>(
+                            new LoadableTypeResolver<>(
+                                    DOWNSTREAM_LINEAGE_TYPE,
+                                    (env) -> ((Dataset) env.getSource()).getUrn()))
+                    )
+            )
+            .type("Owner", typeWiring -> typeWiring
+                    .dataFetcher("owner", new AuthenticatedResolver<>(
+                            new LoadableTypeResolver<>(
+                                    CORP_USER_TYPE,
+                                    (env) -> ((Owner) env.getSource()).getOwner().getUrn()))
+                    )
+            )
+            .type("RelatedDataset", typeWiring -> typeWiring
+                    .dataFetcher("dataset", new AuthenticatedResolver<>(
+                            new LoadableTypeResolver<>(
+                                    DATASET_TYPE,
+                                    (env) -> ((RelatedDataset) env.getSource()).getDataset().getUrn()))
+                    )
+            );
+    }
+
+    /**
+     * Configures resolvers responsible for resolving the {@link com.linkedin.datahub.graphql.generated.CorpUser} type.
+     */
+    private static void configureCorpUserResolvers(final RuntimeWiring.Builder builder) {
+        builder.type("CorpUserInfo", typeWiring -> typeWiring
+            .dataFetcher("manager", new AuthenticatedResolver<>(
+                    new LoadableTypeResolver<>(
+                            CORP_USER_TYPE,
+                            (env) -> ((CorpUserInfo) env.getSource()).getManager().getUrn()))
+            )
+        );
+    }
+
+    /**
+     * Configures {@link graphql.schema.TypeResolver}s for any GQL 'union' or 'interface' types.
+     */
+    private static void configureTypeResolvers(final RuntimeWiring.Builder builder) {
+        builder
+            .type("Entity", typeWiring -> typeWiring
+                    .typeResolver(new EntityInterfaceTypeResolver(LOADABLE_TYPES.stream()
+                            .filter(graphType -> graphType instanceof EntityType)
+                            .map(graphType -> (EntityType<?>) graphType)
+                            .collect(Collectors.toList())
+                    )))
+            .type("PlatformSchema", typeWiring -> typeWiring
+                    .typeResolver(new PlatformSchemaUnionTypeResolver())
+            );
+    }
+
+    /**
+     * Configures custom type extensions leveraged within our GraphQL schema.
+     */
+    private static void configureTypeExtensions(final RuntimeWiring.Builder builder) {
+        builder.scalar(GraphQLLong);
+    }
```

Comment:
Why do we need a custom long type? Does graphql not support integers out of the box?

Reply:
Integers, yes,. 64-bit longs, no. This is just a safety measure, but beyond Java-land we are at the mercy of JSON + Javascript on the frontend. Do you think I should just use ints?

Reply:
I see. JS cannot support all 64 bit integers natively: https://www.npmjs.com/package/long so it might be better to not encourage it.

Reply:
2038 we are done for!

Reply:
All of the downstream models are giving us longs for time. It is easier to have the ability to return longs instead of integers. Keeping as there is no major downside I see.

Reply:
Ok- sounds good.